java-squid/src/main/java/org/sonar/java/ast/parser/TreeFactory.java

```diff
@@ -1650,52 +1393,35 @@ public ExpressionTree newExpression(JavaTree newToken, Optional<List<AnnotationT
 
   public ExpressionTree completeCreator(Optional<TypeArgumentListTreeImpl> typeArguments, ExpressionTree partial) {
     // TODO typeArguments is a parameterized expression used to chose which constructor to call
-    if (typeArguments.isPresent()) {
-      ((JavaTree) partial).prependChildren(typeArguments.get());
-    }
+    // FIXME SONARJAVA-547 Handle type arguments
     return partial;
   }
 
   public ExpressionTree newClassCreator(TypeTree qualifiedIdentifier, NewClassTreeImpl classCreatorRest) {
-    classCreatorRest.prependChildren((AstNode) qualifiedIdentifier);
     return classCreatorRest.completeWithIdentifier(qualifiedIdentifier);
   }
 
   public ExpressionTree newArrayCreator(TypeTree type, NewArrayTreeImpl partial) {
-    return partial.complete(type,
-      (AstNode) type);
+    return partial.complete(type);
   }
 
   public NewArrayTreeImpl completeArrayCreator(Optional<List<AnnotationTreeImpl>> annotations, NewArrayTreeImpl partial) {
-    if (annotations.isPresent()) {
-      partial.prependChildren(annotations.get());
-    }
+    // FIXME SONARJAVA-547 Handle annotations
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Take the required action to fix the issue indicated by this comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1134)